### ISSUE TITLE
Why doesn't paddle.seed() reset the program.random_seed? 

### ISSUE THREAD

**Issue by Wong4j** [PaddlePaddle/Paddle#55633]
### 请提出你的问题 Please ask your question

I found that dropout's seed is always 0 (from `main_program.random_seed`) instead of the seed I set with `paddle.seed(42)`.
If users set `paddle.seed(42)`, they expect all random seeds in Paddle side (including `program.random_seed`) can be set to 42. But the fact is that they need extra code to set `program.random_seed=42`.

Why is Paddle's seed designed this way?
Would it make more sense to set the `program.random_seed` to the same value as `paddle.seed()` when users create the program?

**Comment by zhiqiu** [comment]
plz give an example

**Comment by Wong4j** [comment]
Example:
```
import paddle
import numpy as np
import paddle.nn.functional as F


paddle.enable_static()
main_prog = paddle.static.default_main_program()
startup_prog = paddle.static.default_startup_program()

main_prog.random_seed = 1
paddle.seed(2)

with paddle.static.program_guard(main_prog, startup_prog):
    x = paddle.static.data(name='x', shape=[10, 10], dtype="float32")
    y = F.dropout(x, 0.1)

_x = np.ones((10, 10), dtype=np.float32)

place = paddle.CUDAPlace(0)
exe = paddle.static.Executor(place)
exe.run(startup_prog)

feed = {"x": _x,}

exe.run(main_prog, feed=feed)
```
Print the `attrs` in this line - [common.py#L1166](https://github.com/PaddlePaddle/Paddle/blob/4f9404a6354f5442f4316c968356dd5aeec69c54/python/paddle/nn/functional/common.py#L1166C13-L1166C19). You will see the `dropout`'s seed is 1, which is set by `main_prog.random_seed = 1` . But I expect it to be set by `paddle.seed(2)`.

**Comment by Wong4j** [comment]
@zhiqiu , I have provided an example. Could you please take a look?

**Comment by Wong4j** [comment]
> plz give an example

Could you please take a look?

**Comment by zhiqiu** [comment]
Basically, `program.random_seed` is a deprecated way to set seed, and you can use `paddle.seed()` only now.
`paddle.seed()` is not only a random seed, but a `random generator`, like [random generator in numpy](https://numpy.org/doc/stable/reference/random/generator.html).

**Comment by Wong4j** [comment]
@zhiqiu Got it. But the `dropout` op still use `program.random_seed`.   Refer to [common.py#L1121](https://github.com/PaddlePaddle/Paddle/blob/4f9404a6354f5442f4316c968356dd5aeec69c54/python/paddle/nn/functional/common.py#L1121C20-L1121C20).
The trouble is that I cannot use `paddle.seed()` to set the seed of the `dropout` op
